luxonis_train/core/core.py

```diff
@@ -429,16 +431,68 @@ def infer(
         @param save_dir: Directory where to save the visualizations. If
             not specified, visualizations will be rendered on the
             screen.
+        @type img_path: Optional[str]
+        @param img_path: Path to the image file or directory for inference.
+            If None, defaults to using dataset images.
         """
         self.lightning_module.eval()
 
+        if img_path:
+            img_path_obj = Path(img_path)
+            if img_path_obj.is_file():
+                self._process_single_image(img_path_obj, view, save_dir)
+            elif img_path_obj.is_dir():
+                self._process_directory_images(img_path_obj, view, save_dir)
+        else:
+            self._process_dataset_images(view, save_dir)
+
+    def _process_single_image(self, img_path: Path, view: str, save_dir: Optional[str | Path]) -> None:
```

Comment:
It might be better to move these methods to `luxonis_train.core.utils.infer_utils`